File: data/java/p1/s/s2.java
/**
 * Student Submission 2
 * Should match: ref2 (enhanced for loop with explicit addition)
 */
public class s2 {
    public static int getTotal(int[] values) {
        int result = 0;
        for (int val : values) {
            result = result + val;
        }
        return result;
    }

    public static void main(String[] args) {
        int[] myArray = { 10, 20, 30, 40, 50 };
        System.out.println("Total: " + getTotal(myArray));
    }
}

File: data/cpp/p1/s/s2.cpp
#include <iostream>
#include <vector>


/**
 * Student Submission 2
 * Should match: ref2 (range-based for loop with explicit addition)
 */
int getTotal(const std::vector<int> &values) {
  int result = 0;
  for (const auto &val : values) {
    result = result + val;
  }
  return result;
}

int main() {
  std::vector<int> myArray = {10, 20, 30, 40, 50};
  std::cout << "Total: " << getTotal(myArray) << std::endl;
  return 0;
}
